src/controllers/newspapers.controller.ts

```diff
@@ -260,7 +261,8 @@ const getPublishDatesDays = catchAsync(async (req: Request, res: Response) => {
     throw new ApiError(httpStatus.BAD_REQUEST, 'Missing data to process the request')
 
   const dateStart = `${year}-${monthNameToNumber[month]}-01`
-  const dateEnd = `${year}-${monthNameToNumber[month]}-31`
+  const lastDayOfCurrentMonth = DataCheck(monthNameToNumber[month], year)
+  const dateEnd = `${year}-${monthNameToNumber[month]}-${lastDayOfCurrentMonth}`
```

Comment:
It would be better probably if we just run the moment function here, also no need for the word `current` (since it's not current it's the chosen month)

```suggestion
  const lastDayOfMonth = moment('${year}-${month}', 'YYYY-MM').daysInMonth()
  const dateEnd = `${year}-${monthNameToNumber[month]}-${lastDayOfMonth }`
```

don't forget to import moment